sdk/vppagent/commit.go

```diff
@@ -23,99 +23,81 @@ const (
 	createConnectionSleep   = 100 * time.Millisecond
 )
 
-// Flush is a VPP Agent Flush composite
-type Flush struct {
-	endpoint.BaseCompositeEndpoint
+// Commit is a VPP Agent Commit composite
+type Commit struct {
 	Endpoint string
 }
 
 // Request implements the request handler
-func (f *Flush) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
-	if f.GetNext() == nil {
-		err := fmt.Errorf("composite requires that there is Next set")
-		return nil, err
-	}
-
-	incomingConnection, err := f.GetNext().Request(ctx, request)
-	if err != nil {
-		return nil, err
+// Provides/Consumes from ctx context.Context:
+//     VppAgentConfig
+//	   Next
+func (f *Commit) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+	ctx = WithConfig(ctx) // Guarantees we will retrieve a non-nil VppAgentConfig from context.Context
+	vppAgentConfig := Config(ctx)
+	if vppAgentConfig == nil {
+		return nil, fmt.Errorf("received empty VppAgentConfig")
 	}
 
-	connectionData, err := getConnectionData(f.GetNext(), incomingConnection, false)
-	if err != nil {
-		return nil, err
-	}
+	logrus.Infof("Sending VppAgentConfig to VPP Agent: %v", vppAgentConfig)
 
-	dataChange := connectionData.DataChange
-	if dataChange == nil {
-		err = fmt.Errorf("received empty DataChange")
+	if err := f.send(ctx, vppAgentConfig); err != nil {
+		logrus.Errorf("Failed to send vppAgentConfig to VPP Agent: %v", err)
 		return nil, err
 	}
-
-	logrus.Infof("Sending DataChange to VPP Agent: %v", dataChange)
-	err = f.send(ctx, dataChange)
-	if err != nil {
-		logrus.Errorf("Failed to send DataChange to VPP Agent: %v", err)
-		return nil, err
+	if endpoint.Next(ctx) != nil {
+		return endpoint.Next(ctx).Request(ctx, request)
 	}
-
-	return incomingConnection, nil
+	return request.GetConnection(), nil
 }
 
 // Close implements the close handler
-func (f *Flush) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
-	connectionData, err := getConnectionData(f.GetNext(), connection, false)
-	if err != nil {
-		return &empty.Empty{}, err
-	}
+// Provides/Consumes from ctx context.Context:
+//     VppAgentConfig
+//	   Next
+func (f *Commit) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
```

Comment:
Seems all /sdk/vppagent composites (except `Commit`) have almost identical Request() and Close() methods. Can we fix this somehow? Maybe can we store flushed data in a map inside `Commit` and just get it from it on Close()?

Reply:
Interesting idea :)